tests/test_io_devices.py

```diff
@@ -0,0 +1,198 @@
+from .pigpio_mocks import *
+import vent.io.devices as iodev
+from socket import error as socket_error
+
+
+def test_mock_pigpio_base(patch_pigpio_base):
+    """__________________________________________________________________________________________________________TEST #1
+        Tests that the base fixture of pigpio mocks is indeed enough to instantiate pigpio.pi()
+    """
+    pig = iodev.PigpioConnection()
+    assert isinstance(pig, pigpio.pi)
+    assert pig.connected
+    pig.stop()
+    assert not pig.connected
+    """__________________________________________________________________________________________________________
+    """
+
+
+def test_pigpio_connection_exception(patch_pigpio_base, monkeypatch):
+    """__________________________________________________________________________________________________________TEST #2
+        Tests to make sure an exception is thrown if, upon init, a PigpioConnection finds it is not connected."""
+    def mock_create_bad_connection(host, timeout):
+        """ mock of socket.create_connection(). Returns a bare-bones mock socket"""
+        raise socket_error
```

Comment:
Uh, no, but it could - I guess I didn't think making another fixture for one test was worth it, but maybe it'll be useful for some future thing so why not

Reply:
gotcha, just wasn't sure where `socket_error` came from

Reply:
It does now